cmd/kubectl-gadget/main.go

```diff
@@ -2,25 +2,26 @@ package main
 
 import (
 	"os"
+	"strings"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
 )
 
 var rootCmd = &cobra.Command{
 	Use:   "kubectl-gadget",
 	Short: "Collection of gadgets for Kubernetes developers",
 }
 
+var (
+	KubernetesConfigFlags = genericclioptions.NewConfigFlags(false)
+)
+
 func init() {
 	cobra.OnInitialize(cobraInit)
-
-	// add kubeconfig flag
-	rootCmd.PersistentFlags().String(
-		"kubeconfig",
-		"",
-		"Path to kubeconfig file")
-	viper.BindPFlag("kubeconfig", rootCmd.PersistentFlags().Lookup("kubeconfig"))
+	KubernetesConfigFlags.AddFlags(rootCmd.PersistentFlags())
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
```

Comment:
Do we have any flag with a `-` or environment variable with a `_`?

Reply:
There are some flags from the `genericclioptions` package with `-` like `as-group`, `certificate-authority`, `client-certificate`, etc.
It was suggested to be used in the plugin example.